synapse/handlers/room_member.py

```diff
@@ -188,16 +188,6 @@ async def _local_membership_update(
             require_consent=require_consent,
         )
 
-        # Check if this event matches the previous membership event for the user.
```

Comment:
Is there any concern that removing this check from earlier on might cause odd errors (e.g. rate limiting or in `send_membership_event` it could cause auth errors) to be hit in the case that a duplicate event is being persisted?

Reply:
In this case, the only difference between doing it here and at the start of `handle_new_client_event` is the "newly joined" code between lines 204 and 223, which by definition won't do anything in the case of a duplicate state event.

In `send_membership_event`, we'll now be doing auth checks on duplicate events where we previously would have skipped them. It's *possible* that a duplicate event will fail such checks, but that's a bit of an edge case and I don't really think it matters: fundamentally it suggests the client is trying to do something that the server doesn't want to allow.